### PR TITLE
Draft: Add API method to create scripting environment 

### PR DESCRIPTION
**Pull request by Saturn745**
This adds a new API method, `LuaLinkAPI#createNewScriptEnvironment`, which will create a similar Lua environment used in LuaLink but for plugins to use externally. This could be useful for plugins that want to add some sort of Lua scripting but the use of addons doesn't fit, such as wanting to modify how scripts load, what they can do, their globals, and more. I could see that being particularly useful for something like a minigames plugin that would want to implement the core as a plugin but provide Lua scripting for the minigames themselves. `LuaLink-Projects` is another plugin which I want to do in the near feature which would provide a "project" based approach rather than single scripts like now (Such "projects" feature may be added directly into LuaLink however for now it will be a separate plugin.

TL;DR: Adds a API method to create a similar scripting envoriment to the one used in LuaLink scripts so other plugins can add their own scripting.

###### _This code is a bit of a mess.._

### CODE REVIEW

**Comment by Saturn745** [comment]
Will be implemented along with #43 instead
